git_machete/cmd.py

```diff
@@ -1616,10 +1685,16 @@ def is_merged_to(cli_ctxt: CommandLineContext, b: str, target: str) -> bool:
         # (reflog stripped of trivial events like branch creation, reset etc.)
         # is non-empty.
         return bool(filtered_reflog(cli_ctxt, b, prefix="refs/heads/"))
-    else:
+    elif cli_ctxt.opt_merge:
+        # In "merge" mode.
         # If a branch is NOT equal to the target (typically its parent),
         # it's just enough to check if the target is reachable from the branch.
         return is_ancestor(cli_ctxt, b, target)
```

Comment:
Hmm so in non-merge mode, we don't want to rely on `is_ancestor` completely?

....oh okay, that's why you need check the entire `earlier_sha..later_sha` in https://github.com/VirtusLab/git-machete/pull/129/files#r642599762 right? 💡

Reply:
Coz if you were comparing the tree just against `later_sha` and NOT its ancestor commits, then if `later_sha` already had more commits added after `earlier_sha` was merged, we'd say that _nope, earlier_sha is NOT merged to later_sha _, right?